### DISCUSSION TITLE
Fork with Cyrus-imap

### DISCUSSION THREAD

**Discussion by vid** [docker-mailserver/docker-mailserver#4217] [Ideas]
Hi, I've been running docker-mailserver for a while, and it's been great. However, several features of cyrus would be useful to me (calendar, contacts servers, jmap support). Would it make sense for there to be a fork of this project that uses cyrus instead of dovecot? I could help with testing, but I haven't used cyrus in any advanced way so it would really need some cyrus expertise.

Thanks!

**Comment by polarathene** [comment]
Usually DMS is quite flexible that you probably don't have to maintain a fork.

See the [feature contribution section of my comment here](https://github.com/docker-mailserver/docker-mailserver/pull/4197#issuecomment-2375873666) where it shows how you can extend the DMS image to add any packages you need, and include additional services to run during container startup.

You would use `SMTP_ONLY=1` to disable Dovecot, then follow our docs for `postfix-main.cf` to override / add settings for Postfix `main.cf` to integrate with Cyrus. [Here is the `SMTP_ONLY=0` (default, enable Dovcot) integration looks like for Postfix](https://github.com/docker-mailserver/docker-mailserver/blob/26a44995a9bba825b1348dbf3551687c2100f7a3/target/scripts/startup/setup.d/postfix.sh#L71-L85).

If you'd like to share that for other users interested to benefit, you can do so here in your discussion or contribute an example to our docs 👍
